### PR TITLE
ref(derived_code_mappings): Minor touches

### PR DESCRIPTION
**Pull request by armenzg**

### CODE REVIEW

File: src/sentry/integrations/github/client.py [hunk 1]
```diff
@@ -97,20 +97,20 @@ def get_tree(self, repo_full_name: str, tree_sha: str) -> List[str]:
             if msg == "Git Repository is empty.":
                 logger.warning(f"{repo_full_name} is empty.")
             elif msg == "Not Found":
-                logger.error(f"The Github App does not have access to {repo_full_name}.")
```
Comment (armenzg) [inline src/sentry/integrations/github/client.py:100]: Calling `logger.error` actually creates Sentry issues. I'm hoping `logger.warning` does not.

File: src/sentry/integrations/github/client.py [hunk 2]
```diff
@@ -97,20 +97,20 @@ def get_tree(self, repo_full_name: str, tree_sha: str) -> List[str]:
...
+                logger.warning(f"The Github App does not have access to {repo_full_name}.")
             else:
-                sentry_sdk.capture_exception(e)
```
Comment (armenzg) [inline src/sentry/integrations/github/client.py:102]: `logger.exception` is our preferred way of reporting an exception.

File: src/sentry/integrations/github/client.py [hunk 3]
```diff
@@ -97,20 +97,20 @@ def get_tree(self, repo_full_name: str, tree_sha: str) -> List[str]:
...
+                logger.exception("An unknown error has ocurred.")
 
         return tree
 
     def get_trees_for_org(
-        self, org_slug: str, gh_org: str, cache_seconds: int = 3600 * 24
```
Comment (armenzg) [inline src/sentry/integrations/github/client.py:107]: I'm writing a script that is not tied to an org, thus, naming it `cache_key` is more generic.

File: src/sentry/tasks/derive_code_mappings.py [hunk 4]
```diff
@@ -43,6 +43,7 @@ def derive_code_mappings(
     set_tag("organization.slug", organization.slug)
     # When you look at the performance page the user is a default column
     set_user({"username": organization.slug})
+    set_tag("project.slug", project.slug)
```
Comment (armenzg) [inline src/sentry/tasks/derive_code_mappings.py:46]: Setting the tag sooner means that we increase the chances of an error containing the tag (e.g. if an error occurred before reaching `report_project_codemappings`).

File: src/sentry/integrations/github/client.py (hunk 1 above)
Comment (ceorourke) [reply src/sentry/integrations/github/client.py:100]: I think that's correct

**Review by ceorourke — APPROVED** [review]

File: src/sentry/integrations/github/client.py (hunk 1 above)
Comment (armenzg) [reply src/sentry/integrations/github/client.py:100]: A [sample issue](https://sentry.io/organizations/sentry/issues/3727674865/?project=1&query=is%3Aunresolved+derive+level%3Aerror&referrer=issue-stream&statsPeriod=14d) for the curious:
image

**Pull request merged**